core/src/main/java/io/questdb/griffin/ExpressionParser.java

```diff
@@ -1311,6 +1359,16 @@ void parseExpr(GenericLexer lexer, ExpressionParserListener listener, SqlParserC
                 argStackDepth = onNode(listener, node, argStackDepth, caseCount == 0);
             }
 
+            if (shadowParseMismatchFirstPosition != -1) {
+                // todo (sivukhin): Can we add full query here? Or maybe some trace id?
+                // todo (sivukhin): Also, we can add link to the doc/migration guide
```

Comment:
> You can paste example queries as docstrings, the kind of queries you'd run in tests

My first todo comment were about log entry: currently I log only expression part (so `WHERE` clause for example) - but user will not see full query in the log (which can have table name for example - and this can help user to identify source of these queries).

It's a bit non-trivial thing to do - as full query parsing is even more complicated part compared to `ExpressionParser`and I'm afraid that a lot of code will be touched in order to show full query instead of broken clause...

Reply:
Yes, I understand.

> Also, we can add link to the doc/migration guide

Regarding this - I agreed this is not necessary.

> Can we add full query here? Or maybe some trace id?

But here I don't know what to do. You suggest just to leave logging as is, right? There is no way to log some trace id maybe (do `QuestDB` have something like that) or get access to the full query instead of only expression clause?

Reply:
> Can a user be confident that everything is fine if this log never appears?

Actually, this can be done. Current version in the PR handles almost all cases except `NOT` operator. But I guess we can make this work too - and then it will be the guarantee: if no warning of this format appeared in the logs - then there were no queries with "ambiguous" parsing in the system (for sure, there can be a query which will happen once in a month - but it's out of scope of this guarantee). 

> Why should they rely on a log message (which might be disabled in production) for safety?

I'm actually think that this is fine to some extent. Me, as an external user of several databases, look at the logs as to an additional database "instrument". It has very vague API - but I can use it (with `grep` for example) and analyze to some extent. So I think this is fine to use logs in some case of breaking changes (like this). Plus, I guess `QuestDB` already report some warnings in the log (for example, deprecated parameters) - don't see any difference with our situation here.

> Imo, its false security and users should review the situation themselves and not rely on a log message to determine whether they have a problem or not

I think I disagree here too. It seems to me that with this kind of arguments everything can be called as "false security" and finally you end up forcing users to read all patches to `QuestDB` / `JVM` / `Linux kernel` / etc. 

Also, as I sad it earlier, from the user perspective I find it very unsatisfying to have migration plan which includes step "review all your queries to not have expression with ambiguous binary operator parsing". If I were maintainer of `QuestDB` installation - I may even not know "all sources" of queries and in medium/large-size org this migration can spans months or even more (this obviously based on some my experience - not `QuestDB` - just general SWE experience). And this is kind of tedious work to find all usages, validate them manually and fix them. 

So, I think it's really important to provide more smooth migration path which will help users to adopt new version more easily. Sure, there are users who disable logs for production cluster - but they should understand that this can hurts them at some point (for example, if something will went wrong with their `QuestDB` instances - there will be only limited support for their issue from the development/support team and this is totally fine; if they disable logs, `QuestDB` should not stop writing them at all - because many users still have logs enabled for their production clusters I guess).

---

Definitely, more opinions will be helpful here (I feel that we are going in circles and can't convince each other - so need more people to have quorum over final decision). 

Not sure about second reviewer - maybe it's better to bring this discussion in "more public" (compared to the PR comments) space independently from review? @nwoolmer, what do you think?

Reply:
Ok, removed todos from here. Leaved log as is - not sure that queryId will be actually helpful...